Common/TableProducer/occupancyTableProducer.cxx

```diff
@@ -0,0 +1,1053 @@
+      }
+      hasCollision = false;
+      isAmbgTrack = false;
+      if (track.collisionId() < 0 || collisions.size() <= track.collisionId()) {
+        if (track.collisionId() < 0) {
+          CollisionId_negError++;
+          // check if it is an ambiguous track
+          int ambgPos = FindInTable(track.globalIndex(), ambgTracks);
+          if (ambgPos >= 0 && (ambgTracks.iteratorAt(ambgPos).trackId() == track.globalIndex())) {
+            nAmbgTracks++;
+            isAmbgTrack = true;
+          } else {
+            LOG(error) << "DEBUG :: ERROR :: Not an ambiguous track either ::";
+          }
+        } else {
+          CollisionId_posError++;
+        }
+      } else {
+        CollisionId_NoError++;
+        hasCollision = true;
+      }
+
+      if (!hasCollision && !isAmbgTrack) {
+        LOG(error) << "DEBUG :: ERROR :: A track with no collsiion and is not Ambiguous";
+      }
+      if (hasCollision && isAmbgTrack) {
+        LOG(error) << "DEBUG :: ERROR :: A track has collision and is also ambiguous";
+      }
+
+      if (hasCollision) {
+        LastTrackHadCollision = true;
+      }
+      doCollisionUpdate = false; // default is false;
+      doAmbgUpdate = false;
+      if (hasCollision) {
+        if (LastTrackHadCollision) {
+          if (collision.globalIndex() == oldCollisionIndex) { // if collisions are same
+            doCollisionUpdate = false;
+          } else {
+            doCollisionUpdate = true;
+          } // if collisions are different
+        } else {
+          doCollisionUpdate = true;
+        } // LastTrackWasAmbiguous
+      } else if (isAmbgTrack) {
+        doAmbgUpdate = true;
+        // To be updated later
+        //  if(LastTrackIsAmbg){
+        //    if( haveSameInfo ) { doAmbgUpdate = false;}
+        //    else              { doAmbgUpdate = true; }
+        //  }
+        //  else { doAmbgUpdate = true;} //Last track had Collisions
+      }
+
+      if (doAmbgUpdate) {
+        continue;
+      }
+      if (doCollisionUpdate || doAmbgUpdate) { // collision.globalIndex() != oldCollisionIndex){ //don't update if info is same as old collision
+        if (doCollisionUpdate) {
+          oldCollisionIndex = collision.globalIndex();
+          bc = collision.bc_as<myBCTable>();
+        }
+        if (doAmbgUpdate) {
+          // to be updated later
+          //  bc = collisions.iteratorAt(2).bc_as<aod::BCsWithTimestamps>();
+          //  bc = ambgTracks.iteratorAt(0).bc_as<aod::BCsWithTimestamps>();
+        }
+        // LOG(info)<<" What happens in the case when the collision id is = -1 and it tries to obtain bc"
+        GetTimingInfo(bc, time, TFidThis, bcInTF);
+      }
+
+      if (TFidThis != oldTFid) {
+        oldTFid = TFidThis;
+        auto OccList = Occs.iteratorAt(bc.occId());
```

Comment:
The index dereference should be used directly

Reply:
It didn't work for join of index table and required table, so sticking to iteratorAt() for now